cmd/snap-bootstrap/initramfs_systemd_mount.go

```diff
@@ -71,39 +206,22 @@ type systemdMountOptions struct {
 	// the file system.
 	NoDev bool
 	// NoExec indicates to not allow direct execution of any binaries on the
-	// mounted file system
+	// mounted file system.
 	NoExec bool
-	// Bind indicates a bind mount
+	// Bind indicates a bind mount.
 	Bind bool
 	// Read-only mount
 	ReadOnly bool
 	// Private mount
 	Private bool
 	// Umount the mountpoint
 	Umount bool
-	// Overlayfs indicates an overlay filesystem.
-	Overlayfs bool
-	// Directories to be used as lower layers of an overlay mount.
-	// It does not need to be on a writable filesystem.
-	LowerDirs []string
-	// A directory to be used as the upper layer of an overlay mount.
-	// This is normally on a writable filesystem.
-	UpperDir string
-	// A directory to be used as the workdir of an overlay mount.
-	// This needs to be an empty directory on the same filesystem as upperdir.
-	WorkDir string
-	// dm-verity hash device
-	VerityHashDevice string
-	// dm-verity root hash
-	VerityRootHash string
-	// dm-verity hash offset. Need to be specified if only verity data are
-	// appended to the snap. Defaults to 0 in mount command
-	VerityHashOffset uint64
+	// OverlayFsOpts groups the options related to overlayfs mounts.
```

Comment:
Should this then become:
```
fsOpts interface {  AppendOptions() ([]string, error) }
```

I suggested something like this in the original PR that introduced this code and I'm very happy to see that we are very close to that.

Unless I'm missing a case where you'd want both at the same time, as the code currently allows.

Reply:
That is right, I couldn't think of a case where I would need to supply both at the same time, so I changed this too as suggested. Thanks.